### ISSUE TITLE
StringIndexOutOfBoundsException in FrameParser

### ISSUE THREAD

**Issue by ernestojpg** [pmlopes/mod-stomp-io#8]
I found an uncaught StringIndexOutOfBoundsException in the FrameParser class:

```
Exception in Java verticle
java.lang.StringIndexOutOfBoundsException: String index out of range: -1
	at java.lang.String.substring(String.java:1954)
	at com.jetdrone.vertx.mods.stomp.FrameParser.parseResult(FrameParser.java:51)
	at com.jetdrone.vertx.mods.stomp.FrameParser.handle(FrameParser.java:146)
	at com.jetdrone.vertx.mods.stomp.StompClient$4.handle(StompClient.java:254)
	at com.jetdrone.vertx.mods.stomp.StompClient$4.handle(StompClient.java:249)
	at org.vertx.java.core.net.impl.DefaultNetSocket.handleDataReceived(DefaultNetSocket.java:267)
	at org.vertx.java.core.net.impl.VertxNetHandler.channelRead(VertxNetHandler.java:48)
	at org.vertx.java.core.net.impl.VertxNetHandler.channelRead(VertxNetHandler.java:32)
	at org.vertx.java.core.net.impl.VertxHandler.channelRead(VertxHandler.java:155)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:332)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:318)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:787)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:125)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:507)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:464)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:378)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:350)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:116)
	at java.lang.Thread.run(Thread.java:745)
```

It can be reproduced easily connecting to a port with a non Stomp-compliant server. For example, try to connect to ActiveMQ 61616 port (rather to the standard stomp 61613 port).

I think it can be easily solved catching a more generic `IndexOutOfBoundsException` in the public `handle()` method, rather than a specific `ArrayIndexOutOfBoundsException`.

**Comment by pmlopes** [comment]
Merged and released 1.1.2 to maven central as i write...

**Comment by ernestojpg** [comment]
Wow! Thank you very much Paulo!
